### ISSUE TITLE
Using pre-request vars to insert numbers or booleans in a JSON body creates an escaped output

### ISSUE THREAD

**Issue by kristofferpeterhansel** [usebruno/bruno#2264]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

When using pre-request vars in a JSON body it works fine if it is used inside a text. But if setting a number or boolean the output body becomes a text escaped to fit in a JSON string instead. Like so: 

` "{\n \"test\": \"some text\", // test\n \"bool-test\": true\n}"`

### .bru file to reproduce the bug

```
meta {
  name: broken - insert boolean
  type: http
  seq: 5
}

post {
  url: https://echo.free.beeceptor.com
  body: json
  auth: none
}

body:json {
  {
    "test": "some text", // test
    "bool-test": {{the-value}}
  }
}

vars:pre-request {
  the-value: true
}
```

### Screenshots/Live demo link

Bruno collection with different examples of this behavior

https://gist.github.com/kristofferpeterhansel/d6a3d9f1c07dbd8a2b92ef768bff912f

**Comment by kristofferpeterhansel** [comment]
> @kristofferpeterhansel 
> 
> 
> 
> I see that the json body has a comment. Could you try again by removing it ?
> 
> 
> 
> should be this:
> 
> 
> 
> ```
> 
> body:json {
> 
>   {
> 
>     "test": "some text",
> 
>     "bool-test": {{the-value}}
> 
>   }
> 
> }
> 
> ```

You are right. I didn't test that permutation. That does work. 

But interestingly in the linked examples I have a working example with a comment when the inserted value is a string.

**Comment by thubble** [comment]
I think this is a duplicate of #888

**Comment by kristofferpeterhansel** [comment]
> I think this is a duplicate of #888

Yeah. You might be right. I wasn't searching for the comment angle when I created this.

**Comment by kristofferpeterhansel** [comment]
Looks like this was fixed in 1.20

**Comment by kaitlynforks** [comment]
If you landed here because of https://github.com/usebruno/bruno/discussions/3072:
"With the release of v1.28.0, we’ve shifted to treating all Request Variables as strings instead of inferring the data type from the value. See discussion for more information."

Here is a workaround for Non-String Values in Dynamic Data Testing
Since Bruno doesn’t natively support numbers or booleans in JSON files currently, you can implement a workaround by manually casting values to your request body in your pre request script, transforming the string back into its intended type.
For example:

// Convert string to int in GraphQL input request body
const capturedNum = parseInt(bru.getVar('jsonNum'), 10);
req.body.variables.input.reqNum = capturedNum;

// Convert string to boolean in request body
const capturedBool = bru.getVar('jsonBool');
req.body.variables.input.reqBool = capturedBool;
(I am doing this on GraphQL mutations)

**Comment by kaitlynforks** [comment]
If you landed here because of [Bruno Issue #3072](https://github.com/usebruno/bruno/discussions/3072):  
"With the release of v1.28.0, we’ve shifted to treating all Request Variables as strings instead of inferring the data type from the value. See discussion for more information."

### **Here is a workaround for Non-String Values in Dynamic Data Testing**

Since Bruno doesn’t natively support numbers or booleans in JSON files currently, you can implement a workaround by manually casting values to your request body in your pre request script, transforming the string back into its intended type.
For example:

```javascript
// Convert string to int in GraphQL input request body
const capturedNum = parseInt(bru.getVar('jsonNum'), 10);
req.body.variables.input.reqNum = capturedNum;

// Convert string to boolean in request body
const capturedBool = bru.getVar('jsonBool');
req.body.variables.input.reqBool = capturedBool;
```

(I am doing this on GraphQL mutations)
